src/main/java/org/springframework/data/couchbase/repository/CouchbaseRepository.java

```diff
@@ -32,15 +48,135 @@
 @NoRepositoryBean
 public interface CouchbaseRepository<T, ID> extends PagingAndSortingRepository<T, ID> {
 
+	// all the inherited methods need to be supplemented by ones that take Options, CollectionName and ScopeName args
+
+	@Override
+	Optional<T> findById(ID id);
+	Optional<T> findById(ID id, QueryOptions... options);
+	Optional<T> findById(ID id, QueryOptions options, CollectionName collectionName, ScopeName... scopeNames);
+	Optional<T> findById(ID id, CollectionName collectionName, ScopeName... scopeNames);
+
+	@Override
+	List<T> findAll();
+	List<T> findAll(QueryOptions... options);
+	List<T> findAll(QueryOptions options, CollectionName collectionName, ScopeName... scopeNames);
+	List<T> findAll(CollectionName collectionName, ScopeName... scopeNames);
+
+
+	List<T> findAll(org.springframework.data.couchbase.core.query.Query query);
```

Comment:
Introducing `Query` makes the Repository somewhat pointless as this method duplicates what's possible on the Template API level.